### PR TITLE
Fix gradient checkpointing bug marian

### PR DESCRIPTION
**Pull request by KMFODA**
This PR fixes a bug that a user can encounter while using generate and models that use gradient_checkpointing.

Fixes Issue https://github.com/huggingface/transformers/issues/21737

cc @younesbelkada or @gante

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for fixing! 💯

File: src/transformers/models/marian/modeling_marian.py [hunk 1]
```diff
@@ -995,6 +995,13 @@ def forward(
 
         hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning(
```
Comment (sgugger) [inline src/transformers/models/marian/modeling_marian.py:1000]:
```suggestion
                logger.warning_once(
```

**Review by sgugger — APPROVED** [review]
Thanks for your PR, note that the warning was just changed, so you need a slight edit. :-)
